tests/api2/interfaces.py

```diff
@@ -10,29 +10,134 @@
-from functions import GET
-from config import *
+from functions import GET, PUT, POST, DELETE
 
-BRIDGENETMASKReason = "BRIDGENETMASK not in ixautomation.conf"
+broadcast = ip.replace(ip.split('.')[3], '255')
+aliases = {'address': ip, 'broadcast': broadcast}
 
 
 def test_01_get_interfaces_name():
     results = GET(f'/interfaces?name={interface}')
     assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list) is True, results.text
     assert results.json()[0]["name"] == interface, results.text
 
 
-def test_02_get_interfaces_ip():
+def test_02_get_interfaces_aliases_ip():
     results = GET(f'/interfaces?name={interface}')
     assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list) is True, results.text
     interface_ip = results.json()[0]['state']['aliases'][1]['address']
     assert interface_ip == ip, results.text
 
 
-@pytest.mark.skipif("BRIDGENETMASK" not in locals(),
-                    reason=BRIDGENETMASKReason)
-def test_03_get_interfaces_netmask():
+def test_03_get_interfaces_aliases_broadcast_ip():
     results = GET(f'/interfaces?name={interface}')
     assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list) is True, results.text
+    broadcast_ip = results.json()[0]['state']['aliases'][1]['broadcast']
+    assert broadcast_ip == broadcast, results.text
+
+
+def test_04_get_interfaces_aliases_type():
+    results = GET(f'/interfaces?name={interface}')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list) is True, results.text
+    broadcast_ip = results.json()[0]['state']['aliases'][1]['type']
+    aliases['type'] = broadcast_ip
+    assert isinstance(broadcast_ip, str) is True, results.text
+
+
+def test_05_get_interfaces_aliases_netmask():
+    results = GET(f'/interfaces?name={interface}')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list) is True, results.text
     netmask = results.json()[0]['state']['aliases'][1]['netmask']
-    assert netmask == int(BRIDGENETMASK), results.text
+    aliases['netmask'] = netmask
+    assert isinstance(netmask, int) is True, results.text
+
+
+def test_06_set_main_interfaces_ipv4_to_false():
+    payload = {'ipv4_dhcp': False}
+    results = PUT(f'/interfaces/id/{interface}/', payload)
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict) is True, results.text
+    assert results.json()['ipv4_dhcp'] is False, results.text
+
+
+def test_07_looking_main_interfaces_ipv4_dhcp_if_is_false():
+    results = GET(f'/interfaces/id/{interface}')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict) is True, results.text
+    assert results.json()['ipv4_dhcp'] is False, results.text
+
+
+def test_08_creating_vlan1_interface():
+    global payload
+    payload = {
+        "ipv4_dhcp": True,
+        "name": "vlan2",
```

Comment:
vlan2 and tag 1 is unusual.
You better drop `name`, it should create "vlan1" automatically.

Reply:
I did that to compare name with results output in test 09 and 11. and the name was valn1 have forgotten to put it back to vlan1.